simplipy/api.py

```diff
@@ -259,60 +293,85 @@ def _save_token_data_from_response(self, token_data: dict[str, Any]) -> None:
             self.refresh_token = refresh_token
 
     @staticmethod
-    def is_fatal_error(err: ClientResponseError) -> bool:
+    def is_fatal_error(retriable: list) -> Callable:
         """Determine whether a ClientResponseError is fatal and shouldn't be retried.
 
-        In general, we retry anything outside of HTTP 4xx codes (client errors) with a
-        few exceptions:
+        This call returns the check function.  The arguments to this call are the
+        list of status codes that are retriable.  Generally you'd want to pass
 
         1. 401: We catch this, refresh the access token, and retry the original request.
         2. 409: SimpliSafe base stations regular synchronize themselves with the API,
                 which is where this error can occur; we can't control when/how that
                 happens (e.g., we might query the API in the middle of a base station
                 update), so it should be viewed as retryable.
 
+        But when fetching media files:
+
+        3. 404: When fetching media files, you may get a 404 if the media file is not
+                yet available to read.  Keep trying however, and it will eventually
+                return a 200.
+
         Args:
-            err: An ``aiohttp`` ``ClientResponseError``
+            retriable: A list of retry-able error status codes.
 
         Returns:
-            Whether the error is a fatal one.
+            A callable function used by backoff to check for errors.
         """
-        if err.status in (401, 409):
-            return False
-        return 400 <= err.status < 500
+
+        def check(err: ClientResponseError) -> bool:
+            """The actual check function used by backoff.
+
+            Args:
+                err:       An ``aiohttp`` ``ClientResponseError``
+
+            Returns:
+                Whether the error is a fatal one.
+            """
+            if err.status in retriable:
+                return False
+            return 400 <= err.status < 500
+
+        return check
 
     def _wrap_request_method(
-        self, request_retries: int
+        self, request_retries: int, giveup_func: Callable, request_func: Callable
```

Comment:
Can we make `Callable` more specific?

Reply:
This was a bit more difficult, but I think I managed it.  There is an internal call to get media content that is wrapped in retry,refresh logic that returns a dict, and a public method that returns the bytes from the dict.  This makes the type hints work out better on the wrapper function.